swift/Workflow/Sources/RenderContext.swift

```diff
@@ -75,6 +75,10 @@ public class RenderContext<WorkflowType: Workflow>: RenderContextType {
         fatalError()
     }
 
+    public func runSideEffect(key: AnyHashable, action: (Lifetime) -> Void) {
```

Comment:
Why `AnyHashable` instead of making the function generic?

```swift
public func runSideEffect<Key: Hashable>(key: Key, action: (Lifetime) -> Void) {
```

Reply:
Either works. 

Not entirely sure how `AnyHashable` works, but seems like it's able to wrap any `Hashable` type without us having to explicitly do the wrapping.

Reply:
#TIL

I assume it has something to do with [this](https://github.com/apple/swift/blob/master/stdlib/public/core/AnyHashable.swift#L284-L288).